### DISCUSSION TITLE
CRM 2016 screens  are not displaying ribbon and command bars options using cypress

### DISCUSSION THREAD

**Discussion by hari521-git** [cypress-io/cypress#14383] [Questions and Help]
Using cypress CRM 2016 screens are not loading fully , not displaying ribbon and command bars to select options. Can some one confirm whether cypress supports CRM 2016 .?

**Comment by aligneddev** [comment]
@hari521-git do you have some error outputs from Cypress when it runs against CRM? Does it load part of the page?
Is this Microsoft CRM?

**Reply by hari521-git** [reply]
@aligneddev Yes Microsoft CRM 2016 pages, it keep loading and not showing ribbon and command bar options and no errors or log details found in cypress steps, step fails after defaultcommndtimeout

**Reply by aligneddev** [reply]
for me, I have "chromeWebSecurity": false in my cypress.json
Setting the ADFS header cookies seems to help after login.
```
Cypress.Cookies.defaults({
    preserve: ['MSISAuth', 'MSISAuth1', 'MSISIPSelectionPersistent', 'MSISAuthenticated', 'MSISSignOut',
      'MSISAuthenticated', 'FedAuth', 'FedAuth1', 'ASP.NET_SessionId', 'MSISLoopDetectionCookie']
  });
```
However, I'm seeing problems with waiting for the loading to complete and getting this message after opening our ribbon bar (I don't know how custom this is vs the default setup for CRM)

```
  it('Open CRM Sales Tab', () =>{
    cy.get('#homeButtonImage', { timeout: 100000 }).click();
    cy.get('#SFA > .navActionButtonIconContainer > .navActionButtonIcon',  { timeout: 100000, force: true }).click();
    cy.get('#chr_claim').click();
  });
```
```
Timed out retrying after 12050ms: cy.click() failed because this element is not visible:

<span class="navActionButtonIcon" unselectable="on">...</span>

This element <span.navActionButtonIcon> is not visible because it has an effective width and height of: 0 x 0 pixels.

Fix this problem, or use {force: true} to disable error checking.Learn more
```

I'm not sure if UI tests against CRM pages is feasible. I think the data could be verified through the CRM OData endpoint and do everything possible to avoid automating the UI in CRM. I guess Selenium would have to be used to do UI automation and clicks in the CRM pages.
